src/3d/qgs3dmapsettings.cpp

```diff
@@ -174,12 +170,7 @@ void Qgs3DMapSettings::readXml( const QDomElement &elem, const QgsReadWriteConte
 
   QDomElement elemTerrain = elem.firstChildElement( QStringLiteral( "terrain" ) );
   mTerrainRenderingEnabled = elemTerrain.attribute( QStringLiteral( "terrain-rendering-enabled" ), QStringLiteral( "1" ) ).toInt();
```

Comment:
what do you think about getting rid of mTerrainRenderingEnabled, and no terrain would be equivalent to setTerrainSettings(nullptr) ? Or do you prefer to just deactivate terrain while keeping the settings, so that they are not lost when terrain is activated again?

Reply:
> Or do you prefer to just deactivate terrain while keeping the settings, so that they are not lost when terrain is activated again

I'm ok with losing the settings. I don't think there's a really strong use case for switching on and off terrain, I think it's more likely a set-once-per-map thing.

Reply:
I prefer not to lose the settings. The cost is quite minimal (some _potentially_ unneeded terrain tags in the xml?) and the GUI suggests you should not lose them (since it's a checkbox, and not a _No Terrain_ entry in the combobox)

Reply:
What if we do make a "no terrain" entry in the combo and drop the checkbox? Are you ok with that?

Reply:
Yes, we would need to do this in that case.
Anyone wanting to toggle the terrain without losing settings will still be able to do it by keeping the settings page open.

Reply:
I'm going to come back to this one -- we'd need to add a lot of nullptr checks against terrainSettings() if we permit that to be null, so I'm going to have a bit of a think about a cleaner way to handle that.